packages/adapter-session-unstorage/src/index.ts

```diff
@@ -0,0 +1,90 @@
+import type {
+	SessionSchema,
+	SessionAdapter,
+	AdapterFunction
+} from "lucia-auth";
+import { type Storage } from "unstorage";
+
+const adapter =
+	(unstorage: {
+		session: Storage;
+		sessionKeyPrefix?: string;
+		userSessionKeyPrefix?: string;
+	}): AdapterFunction<SessionAdapter> =>
+	() => {
+		const {
+			session: sessionUnstorage,
+			sessionKeyPrefix = "session",
+			userSessionKeyPrefix = "user_session"
+		} = unstorage;
+		return {
+			sessionKeyPrefix,
+			userSessionKeyPrefix,
+			getSession: async (sessionId) => {
+				const sessionData = await sessionUnstorage.getItem(
+					`${sessionKeyPrefix}:${sessionId}`
+				);
+				return sessionData as SessionSchema | null;
+			},
+			getSessionsByUserId: async (userId) => {
+				const userKeys = await sessionUnstorage.getKeys(
+					`${userSessionKeyPrefix}:${userId}`
+				);
+				const sessionIds = await Promise.all(
+					userKeys.map((id) => sessionUnstorage.getItem(id) as Promise<string>)
+				);
+				return Promise.all(
+					sessionIds.map(
+						(id) =>
+							sessionUnstorage.getItem(
+								`${sessionKeyPrefix}:${id}`
+							) as Promise<SessionSchema>
+					)
+				);
+			},
+			setSession: async (session) => {
```

Comment:
What happens when the key already exists? Does it override it? If so, this needs to check id the key already exists and throw a `LuciaError`. If not, you need to catch the error and throw a `LuciaError`.

Reply:
Huh. It needs to pass the tests so the check is required. The test shouldn't be altered.

Reply:
That's what I thought, but the tests aren't working with the redis adapter ...

<img width="743" alt="image" src="https://github.com/pilcrowOnPaper/lucia/assets/13395944/9cabe064-e42e-4c7a-908c-5fc42b2b0909">

Reply:
You can ignore the Redis adapter for now, I'll fix it after v2.

Reply:
Oh yeah, that test doesn't work for session adapters. It's a bit hacky, but can you remove the tests when developing it, and leave it there in the PR? Will fix it after v2.

Reply:
This is the redis version : 

```ts
	setSession: async (session) => {
		await Promise.all([
			userSessionRedis.lPush(session.user_id, session.id),
			sessionRedis.set(session.id, JSON.stringify(session), {
				EX: Math.floor(Number(session.idle_expires) / 1000)
			})
		]);
	},
```

and the unstorage version : 

```ts
			setSession: async (session) => {
				await Promise.all([
					sessionUnstorage.setItem(
						`${userSessionKeyPrefix}:${session.user_id}:${Date.now()}`,
						session.id
					),
					sessionUnstorage.setItem(`${sessionKeyPrefix}:${session.id}`, session) //@todo This key should expire automatically.
				]);
			},
```

They have the same behaviour, but unstorage doesn't support expiring keys yet (I've opened a PR to add that feature in their repo)

Reply:
Ah got it, I added a flag to disable these 2 in session adapters.